tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -1223,6 +1271,14 @@ SpirvVariable *DeclResultIdMapper::createStructOrStructArrayVarOfExplicitLayout(
   return var;
 }
 
+SpirvVariable *DeclResultIdMapper::createStructOrStructArrayVarOfExplicitLayout(
+    const DeclContext *decl, int arraySize, const ContextUsageKind usageKind,
+    llvm::StringRef typeName, llvm::StringRef varName) {
+  return createStructOrStructArrayVarOfExplicitLayout(
+      decl, arraySize == 0 ? llvm::SmallVector<int, 2>{} : llvm::SmallVector<int, 2>{arraySize},
```

Comment:
I'd just create a local variable for this array. Also, since you have 0 or 1 elements, small size of 1 would be enough.
```c++
llvm::SmallVector<int, 1> arraySizes;
if (arraySize > 0)
  arraySizes.push_back(arraySize);
return createStructOrStructArrayVarOfExplicitLayout(decl, arraySizes, usageKind, typeName, varName);
```